src/strongbox/catalogue.clj

```diff
@@ -30,8 +30,18 @@
     (try
       (if-not (contains? dispatch-map key)
         (error (format "addon '%s' is from source '%s' that is unsupported" (:label addon) key))
-        (when-let [source-updates ((get dispatch-map key) addon game-track)]
-          (merge addon source-updates)))
+        (let [release-list ((get dispatch-map key) addon game-track)
+              latest-release (first release-list)
+              pinned-release (first (filter (fn [release]
```

Comment:
this is `addon/find-release` but for the pinned release rather than the installed release. improve addon/find-release and shift this logic there.

Reply:
test